### PR TITLE
dunst: make icon_path extensible

### PR DESCRIPTION
**Pull request by ncfavier**
### Description

This makes `settings.global.icon_path` a `separatedString ":"` so that the user can add their own paths.

I need this because the current `icon_path` setting is quite flaky (but I don't think there's a better way apart from implementing the icon theme specification, or waiting until dunst does).

### Checklist

- [X] Change is backwards compatible.

- [X] Code formatted with `./format`.

- [X] Code tested through `nix-shell --pure tests -A run.all`.

- [ ] Test cases updated/added. See [example](https://github.com/nix-community/home-manager/commit/f3fbb50b68df20da47f9b0def5607857fcc0d021#diff-b61a6d542f9036550ba9c401c80f00ef).

- [X] Commit messages are formatted like

    ```
    {component}: {description}

    {long description}
    ```

    See [CONTRIBUTING](https://github.com/nix-community/home-manager/blob/master/doc/contributing.adoc#sec-commit-style) for more information and [recent commit messages](https://github.com/nix-community/home-manager/commits/master) for examples.

### CODE REVIEW

File: modules/services/dunst.nix [hunk 1]
```diff
@@ -138,7 +146,7 @@ in {
           "status"
           "stock"
         ];
-      in concatStringsSep ":" (concatMap (theme:
+      in mkMerge (concatMap (theme:
```
Comment (berbiche) [inline modules/services/dunst.nix:149]: Why are you using `mkMerge` here?
`mkMerge` is only relevant to merge multiple, possibly conflicting module configuration.

While at it, can you refactor this to use variables? The current nesting makes it hard to understand what the function does.

File: modules/services/dunst.nix [hunk 2]
```diff
@@ -78,7 +78,15 @@ in {
...
-        type = with types; attrsOf (attrsOf eitherStrBoolIntList);
+        type = types.submodule {
+          freeformType = with types; attrsOf (attrsOf eitherStrBoolIntList);
+          options = {
+            global.icon_path = mkOption {
+              type = types.separatedString ":";
+              description = "Paths where dunst will look for icons.";
+            };
+          };
+        };
```
Comment (berbiche) [inline modules/services/dunst.nix:89]: 👍 for using a freeform module.

**Review by berbiche — CHANGES_REQUESTED** [review]
Thanks for the contribution!

File: modules/services/dunst.nix (hunk 1 above)
Comment (ncfavier) [reply modules/services/dunst.nix:149]: The idea is that the type of the option already encodes the implementation detail that it's a colon-separated list of paths, so the code can abstract that away. Arguably making the type something like "path or list of paths" with the appropriate merge semantics (like what NixOS does for [systemd unit options](https://github.com/NixOS/nixpkgs/blob/nixos-unstable/nixos/modules/system/boot/systemd-unit-options.nix#L18-L28)) would be more elegant, but it's more work.

I'll revert to `concatStringsSep ":"` if you still think `mkMerge` is confusing.

Comment (ncfavier) [reply modules/services/dunst.nix]: > While at it, can you refactor this to use variables? The current nesting makes it hard to understand what the function does.

Is this better?

Comment (berbiche) [reply modules/services/dunst.nix]: Yup, very clean. I had no idea `cartesianProductOfSets` existed.

**Review by berbiche — APPROVED** [review]

**Comment by berbiche** [comment]
Tested locally :+1:

**Comment by ncfavier** [comment]
Thanks!

**Pull request merged**